src/sinks/prometheus/remote_write.rs

```diff
@@ -103,6 +109,57 @@ impl SinkConfig for RemoteWriteConfig {
     }
 }
 
+///
+use crate::sinks::util::batch::{Batch, BatchError, PushResult};
+use std::sync::Arc;
+
+struct EncodeBatch<B, I, O>
+where
+    B: Batch<Output = I>,
+{
+    inner: B,
+    transform: Arc<Box<dyn Fn(I) -> O + Send + Sync>>,
```

Comment:
How much does the `Arc` gain us here over just adding `Clone` to the list of traits? Could the thing in the `Box` be an `Arc` to have the same effect without requiring the `Arc`?

Reply:
Indeed, we can use `Clone` instead `Arc` :+1: